### PR TITLE
2nd Assignment by 202503037 (Jiho Park)

### PR DESCRIPTION
**Pull request by 202503037-code**
# Assignment Submission

## Student Information
- **Student ID**: [Your Student ID]
- **Name**: [Your Full Name]
- **Assignment**: [Assignment Number]

## Changed Files

```
submissions
└── 202503037
    ├── assignment2
    │   ├── README.md
    │   ├── problem1
    │   │   ├── README.md
    │   │   └── main.py
    │   ├── problem2
    │   │   ├── README.md
    │   │   └── textops
    │   │       ├── __init__.py
    │   │       ├── clean
    │   │       │   ├── __init__.py
    │   │       │   └── filters.py
    │   │       ├── main.py
    │   │       └── tokenize
    │   │           ├── __init__.py
    │   │           └── word.py
    │   ├── problem3
    │   │   ├── README.md
    │   │   └── main.py
    │   ├── problem4
    │   │   ├── README.md
    │   │   └── dsops
    │   │       ├── __init__.py
    │   │       ├── main.py
    │   │       ├── split
    │   │       │   ├── __init__.py
    │   │       │   └── train_test.py
    │   │       └── stats
    │   │           ├── __init__.py
    │   │           └── labels.py
    │   ├── problem5
    │   │   ├── README.md
    │   │   └── cachekit
    │   │       ├── __init__.py
    │   │       └── main.py
    │   └── problem6
    │       ├── README.md
    │       └── main.py
    └── practice1
        ├── README.md
        └── main.py
```

### Problem 1
- [ ] Implemented required functionality
- [ ] All tests pass locally

### Problem 2
- [ ] Implemented required functionality
- [ ] All tests pass locally

### Problem 3
- [ ] Implemented required functionality
- [ ] All tests pass locally

### Problem 4 (Optional)
- [ ] Implemented required functionality
- [ ] All tests pass locally

### Problem 5 (Optional)
- [ ] Implemented required functionality
- [ ] All tests pass locally

### Problem 6 (Optional)
- [ ] Implemented required functionality
- [ ] All tests pass locally

## Testing
- [ ] I have run `python grader.py` locally
- [ ] All implemented problems pass their tests
- [ ] I have tested edge cases mentioned in the problem statements

## Additional Notes

---
**Note**: This PR will be automatically validated for:
- Correct directory structure
- Valid Python syntax
- PR title format: `Nth Assignment by {student_id} (Full Name)`
- File organization compliance

### CODE REVIEW

**Comment by hist0613** [comment]
:(

File: submissions/202503037/assignment2/problem2/textops/clean/filters.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      3) collapse all whitespace to single spaces
+      4) remove ASCII punctuation except apostrophes (') and hyphens (-)
+    """
+    s=s.lower()
+    s=re.sub(r"\s+", " ", s)
+    s=s.strip() 
+    keep = {"'", "-"}
+    s = ''.join(ch for ch in s if ch not in set(string.punctuation)-keep)
+    return s
+    raise NotImplementedError
```
Comment (hist0613) [inline submissions/202503037/assignment2/problem2/textops/clean/filters.py:18]: 이 라인이 실행이 안 되니까 문제는 없지만, 지우는게 좋습니다 :)

File: submissions/202503037/assignment2/problem2/textops/tokenize/word.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
+def word_tokens(s: str) -> list[str]:
+    """
+    Split on single spaces; empty/whitespace-only -> [].
+    Assume `s` is already normalized by clean_text.
+    """
+    if not s or s.strip() == "":
+        return []
+    s=s.split(" ")
+    return s
+    raise NotImplementedError
```
Comment (hist0613) [inline submissions/202503037/assignment2/problem2/textops/tokenize/word.py:10]: 여기도

File: submissions/202503037/assignment2/grader.py
Comment (hist0613) [inline submissions/202503037/assignment2/grader.py:1]: 지워주세요~

File: submissions/202503037/practice1/main.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
+import time
+#from tqdm import tqdm
```
Comment (hist0613) [inline submissions/202503037/practice1/main.py:2]: 사용되지 않는 코드는 지워주세요!

**Review by hist0613 — APPROVED** [review]
LGTM
